Common/AnatupleProducer.py

```diff
@@ -0,0 +1,139 @@
+import ROOT
+import numpy as np 
+import Common.BaselineSelection as Baseline
+import Common.Utilities as Utilities
+import Common.ReportTools as ReportTools
+
+def AddVariablesToDf(df, var_list, idx, part_name, part_number, new_vars):
+    for var in var_list:
+        partName = "b"
+        df = df.Define(f"{partName}{part_number}_{var}", f"{part_name}_{var}[{idx}]")
+        new_vars.append(f"{partName}{part_number}_{var}")
+    return df 
+
+
+def createAnatuple(inFile, outFile, period, sample, X_mass, snapshotOptions, isData=0):
+    Baseline.Initialize(True, True)
+    df = ROOT.RDataFrame("Events", inFile)
+
+    df = df.Define("sample", f"static_cast<int>(SampleType::{sample})")
+    df = df.Define("period", f"static_cast<int>(Period::Run{period})") 
+    df = df.Define("X_mass", f"static_cast<int>({X_mass})")
+
+    
+    df = Baseline.RecoLeptonsSelection(df)
+    df = Baseline.RecoJetAcceptance(df)
+    df = Baseline.RecoHttCandidateSelection(df)
+    df = Baseline.RecoJetSelection(df) 
+    df = Baseline.DefineHbbCand(df)  
+    df = df.Define("channelId","static_cast<int>(httCand.channel())")
+    df = df.Define("is_data", f"{isData}") 
+    df = df.Define("lumi", "luminosityBlock")
+    df = df.Define("evt", "event")
+    deepTauScores= ["rawDeepTau2017v2p1VSe","rawDeepTau2017v2p1VSmu",\
+                "rawDeepTau2017v2p1VSjet", "rawDeepTau2018v2p5VSe", "rawDeepTau2018v2p5VSmu",\
+                "rawDeepTau2018v2p5VSjet", "decayMode"] 
+    JetObservables = ["hadronFlavour","partonFlavour"]
+    colToSave = ["event","lumi","run","sample", "period", "X_mass","channelId", "is_data",\
+                "MET_pt", "MET_phi","PuppiMET_pt", "PuppiMET_phi",\
+                "DeepMETResolutionTune_pt", "DeepMETResolutionTune_phi","DeepMETResponseTune_pt", "DeepMETResponseTune_phi"] 
+ 
+   
+    for leg_idx in [0,1]:
+        df = df.Define(f"tau{leg_idx+1}_pt", f"httCand.leg_p4[{leg_idx}].Pt()")
+        df = df.Define(f"tau{leg_idx+1}_eta", f"httCand.leg_p4[{leg_idx}].Eta()")
+        df = df.Define(f"tau{leg_idx+1}_phi", f"httCand.leg_p4[{leg_idx}].Phi()")
+        df = df.Define(f"tau{leg_idx+1}_mass", f"httCand.leg_p4[{leg_idx}].M()")
+        df = df.Define(f"tau{leg_idx+1}_charge", f"httCand.leg_charge[{leg_idx}]")
+        colToSave.append(f"tau{leg_idx+1}_pt")
+        colToSave.append(f"tau{leg_idx+1}_eta")
+        colToSave.append(f"tau{leg_idx+1}_phi")
+        colToSave.append(f"tau{leg_idx+1}_mass")
+        colToSave.append(f"tau{leg_idx+1}_charge")
+        
+        df = df.Define(f"tau{leg_idx+1}_iso", f"RVecF iso;\
```

Comment:
iso should be float not RVecF

Reply:
All observables should be float / int (I mean, not vectors), right?

Reply:
yes